sdk/typespec/typespec_client_core/src/http/pipeline.rs

```diff
@@ -122,27 +131,24 @@ mod tests {
             }
         }
 
-        #[derive(Model, Debug, Deserialize)]
-        #[typespec(crate = "crate")]
+        #[derive(Debug, Deserialize)]
         struct Model {
             foo: i32,
             bar: String,
         }
 
-        let options = ClientOptions {
-            transport: Some(TransportOptions::new_custom_policy(Arc::new(Responder {}))),
-            ..Default::default()
-        };
-        let pipeline = Pipeline::new(options, Vec::new(), Vec::new());
+        // Simulated service method
+        async fn service_method() -> crate::Result<Response<Model>> {
+            let options = ClientOptions {
+                transport: Some(TransportOptions::new_custom_policy(Arc::new(Responder {}))),
+                ..Default::default()
+            };
```

Comment:
Seems `Responder` already exists, but FYI: `azure_core_test` does have a `MockTransport` if you're interested. It has some good features and we can add more over time. I patterned it similar to .NET's, having been on that team and using it a while back.

Reply:
This was largely just reorganization to "show" what a service method is expected to write, so I didn't want to change/rewrite anything here since the test already works as-is